src/operations/query.ts

```diff
@@ -149,6 +151,47 @@ const readRootField = (
   }
 };
 
+export const readFragment = (
+  store: Store,
+  query: DocumentNode,
+  id: string
+): Data | null | void => {
+  const fragments = getFragments(query);
+  const names = Object.keys(fragments);
+  const fragment = fragments[names[0]] as FragmentDefinitionNode;
+  if (fragment === undefined) {
+    return warning(
+      false,
+      'readFragment (...) was called with an empty fragment.\n' +
+        'You have to call it with at least one fragment in your GraphQL document.'
+    );
+  }
+
+  const select = getSelectionSet(fragment);
+  const typeName = getFragmentTypeName(fragment);
+  const entityKey = store.keyOfEntity({ __typename: typeName, id }) as string;
+
+  if (!entityKey) {
```

Comment:
You need to check here for the keys config, so that the error doesn't mention it anymore if it exists :)

Reply:
That shouldn't be needed if we have this right? https://github.com/FormidableLabs/urql-exchange-graphcache/pull/73/files#diff-cfb8e44b6757fd740446e9e4486b73c1L160